Mingw-w64 10.0 makes HAVE_VSNPRINTF tests fail. Until it is resolved, patch
broken code which fails to build.
--- libgfortran/runtime/error.c.orig    2022-06-13 15:26:39.436667000 +0200
+++ libgfortran/runtime/error.c 2022-06-13 15:31:27.776457000 +0200
@@ -136,7 +136,7 @@ estr_writev (const struct iovec *iov, int iovcnt)
 }
 
 
-#ifndef HAVE_VSNPRINTF
+#if !(defined(HAVE_VSNPRINTF) || defined(HAVE_MINGW_SNPRINTF))
 static int
 gf_vsnprintf (char *str, size_t size, const char *format, va_list ap)
 {
